### ISSUE TITLE
Unexpected match with **

### ISSUE THREAD

**Issue by fwinkelbauer** [dazinator/DotNet.Glob#33]
Hi,

I just found an unexpected match when fiddling around with your library. I am not sure if this is a bug, or if my understanding is lacking:

```csharp
var glob = DotNet.Globbing.Glob.Parse("Bumpy/**/AssemblyInfo.cs");

// success - expected
Assert.IsTrue(glob.IsMatch("Bumpy/Properties/AssemblyInfo.cs"));

// failure - unexpected
Assert.IsFalse(glob.IsMatch("Bumpy.Test/Properties/AssemblyInfo.cs"));
```

**Comment by dazinator** [comment]
This is expected because the first bit of your pattern: 'Bumpy/` wont ever match `Bumpy.Text/`. If you want to match all paths beginning with Bumpy, try this:

`Bumpy*/**/AssemblyInfo.cs`

Note the additional * that will match any additional characters in that first Bumpy segment including none.

**Comment by dazinator** [comment]
I have found a quirk. If you set some global options (should not be necessary though) then this behaves as expected:

```
 GlobParseOptions.Default.AllowInvalidPathCharacters = true;

```
This option to allow you to use invalid path characters in a glob pattern is something I added very recently so I expect it has introduced this bug! If you need a temporary work around, you can resort to the above, otherwise I'll get a fix out sometime soon.

**Comment by fwinkelbauer** [comment]
Take your time!

**Comment by dazinator** [comment]
Fixed as of version 1.6.5 - thank you for reporting this.
